cf-custom-resources/lib/env-controller.js

```diff
@@ -0,0 +1,236 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+"use strict";
+
+const aws = require("aws-sdk");
+
+// These are used for test purposes only
+let defaultResponseURL;
+
+/**
+ * Upload a CloudFormation response object to S3.
+ *
+ * @param {object} event the Lambda event payload received by the handler function
+ * @param {object} context the Lambda context received by the handler function
+ * @param {string} responseStatus the response status, either 'SUCCESS' or 'FAILED'
+ * @param {string} physicalResourceId CloudFormation physical resource ID
+ * @param {object} [responseData] arbitrary response data object
+ * @param {string} [reason] reason for failure, if any, to convey to the user
+ * @returns {Promise} Promise that is resolved on success, or rejected on connection error or HTTP error response
+ */
+let report = function (
+  event,
+  context,
+  responseStatus,
+  physicalResourceId,
+  responseData,
+  reason
+) {
+  return new Promise((resolve, reject) => {
+    const https = require("https");
+    const { URL } = require("url");
+
+    var responseBody = JSON.stringify({
+      Status: responseStatus,
+      Reason: reason,
+      PhysicalResourceId: physicalResourceId || context.logStreamName,
+      StackId: event.StackId,
+      RequestId: event.RequestId,
+      LogicalResourceId: event.LogicalResourceId,
+      Data: responseData,
+    });
+
+    const parsedUrl = new URL(event.ResponseURL || defaultResponseURL);
+    const options = {
+      hostname: parsedUrl.hostname,
+      port: 443,
+      path: parsedUrl.pathname + parsedUrl.search,
+      method: "PUT",
+      headers: {
+        "Content-Type": "",
+        "Content-Length": responseBody.length,
+      },
+    };
+
+    https
+      .request(options)
+      .on("error", reject)
+      .on("response", (res) => {
+        res.resume();
+        if (res.statusCode >= 400) {
+          reject(new Error(`Error ${res.statusCode}: ${res.statusMessage}`));
+        } else {
+          resolve();
+        }
+      })
+      .end(responseBody, "utf8");
+  });
+};
+
+/**
+ * Control the optional resources of the environment stack by updating the parameters.
+ *
+ * @param {string} requestType Type of the request.
+ * @param {string} stackName Name of the stack.
+ * @param {string} workload Name of the copilot workload.
```

Comment:
Can we pass an additional input to the lambda: 
```suggestion
 * @param {string[]} envParameters List of parameters from the environment stack to update.
```
This way in the future, if we want to add the workload to a new parameter, say `NLBWorkloads`, we won't have to update the lambda code but instead add an additional input in cloudformation:
```yaml
Properties:
   # ...
   Parameters: ['ALBWorkloads']
```